Modern/wrapperPlugins/src/test/java/org/bonej/wrapperPlugins/ElementFractionWrapperTest.java

```diff
@@ -83,35 +83,38 @@ public void testNullImageCancelsElementFraction() throws Exception {
 			ElementFractionWrapper.class);
 	}
 
-	// TODO Rewrite with a hyper stack
 	@Test
-	public void testResults() throws Exception {
+	public void testResults3DHyperstack() throws Exception {
 		// SETUP
 		final String unit = "mm";
 		final double scale = 0.9;
 		final int cubeSide = 5;
-		final int padding = 1;
-		final int stackSide = cubeSide + padding * 2;
-		final double cubeVolume = cubeSide * cubeSide * cubeSide;
-		final double spaceVolume = stackSide * stackSide * stackSide;
+		final int stackSide = cubeSide + 2;
+		final int expectedSize = 4;
 		final double elementSize = scale * scale * scale;
-		final double[] expectedVolumes = { cubeVolume * elementSize };
-		final double[] expectedTotalVolumes = { spaceVolume * elementSize };
-		final double[] expectedRatios = { cubeVolume / spaceVolume };
-		final double[][] expectedValues = { expectedVolumes, expectedTotalVolumes,
-			expectedRatios };
+		final double cubeVolume = cubeSide * cubeSide * cubeSide * elementSize;
+		final double spaceVolume = stackSide * stackSide * stackSide * elementSize;
+		final double ratio = cubeVolume / spaceVolume;
+		final double[] expectedVolumes = {cubeVolume, 0, 0, cubeVolume};
+		final double[] expectedTotalVolumes = {spaceVolume, spaceVolume, spaceVolume, spaceVolume};
+		final double[] expectedRatios = {ratio, 0, 0, ratio};
+		final double[][] expectedValues = {expectedVolumes, expectedTotalVolumes, expectedRatios};
 		final String[] expectedHeaders = { "Bone volume (" + unit + "³)",
 			"Total volume (" + unit + "³)", "Volume ratio" };
-		// Create an test image of a cuboid
-		final Img<BitType> img = ArrayImgs.bits(stackSide, stackSide, stackSide);
-		Views.interval(img, new long[] { 1, 1, 1 }, new long[] { 5, 5, 5 }).forEach(
-			BitType::setOne);
-		final double[] calibration = { scale, scale, scale };
-		final String[] units = { unit, unit, unit };
-		final AxisType[] axes = { Axes.X, Axes.Y, Axes.Z };
+		// Create an hyperstack Img with a cube at (c:0, f:0) and (c:1, f:1)
+		final Img<BitType> img = ArrayImgs.bits(stackSide, stackSide, stackSide, 2, 2);
```

Comment:
I'm guessing c and f are a colour channel and the time channel, but not immediately clear to me